plugin-process-to-cgroup-bridge/src/transform.rs

```diff
@@ -0,0 +1,187 @@
+                    )
+                }
+            }
+        }
+
+        if self.merge_similar_cgroups {
+            let mut aggregated_cgroup_measurements = aggregate_cgroups_measurements(&mut cgroup_measurements);
+            cgroup_measurements.clear();
+            cgroup_measurements.merge(&mut aggregated_cgroup_measurements);
+        }
+
+        measurements.clear();
+        measurements.merge(&mut kept_measurements);
+        measurements.merge(&mut cgroup_measurements);
+
+        Ok(())
+    }
+}
+
+impl ProcessToCgroupBridgeTransform {
+    pub fn new(
+        processes_metrics_ids: Vec<RawMetricId>,
+        merge_similar_cgroups: bool,
+        keep_processed_measurements: bool,
+
+        #[cfg(test)] proc_path: PathBuf,
+    ) -> Self {
+        Self {
+            processes_metrics_ids,
+            merge_similar_cgroups,
+            keep_processed_measurements,
+
+            #[cfg(test)]
+            proc_path,
+        }
+    }
+
+    fn cgroup_measurement_from_measurement(
+        &self,
+        measurement: &MeasurementPoint,
+    ) -> anyhow::Result<Option<MeasurementPoint>> {
+        if !self.processes_metrics_ids.contains(&measurement.metric) {
+            return Ok(None);
+        }
+        let pid = match extract_process_id_from_measurement(measurement) {
+            Ok(pid) => pid,
+            Err(_) => return Ok(None),
+        };
+        let cgroup_path = self.find_cgroup_path_from_process_id(pid)?;
+        let cgroup_consumer = ResourceConsumer::ControlGroup {
+            path: cgroup_path.into(),
+        };
+        let mut cgroup_measurement = measurement.clone();
+        cgroup_measurement.consumer = cgroup_consumer;
+        Ok(Some(cgroup_measurement))
+    }
+
+    fn find_cgroup_path_from_process_id(&self, pid: u32) -> anyhow::Result<String> {
+        let procfs_cgroup_base_path = self.get_proc_path();
+
+        let procfs_cgroup_filepath = procfs_cgroup_base_path.join(pid.to_string()).join("cgroup");
+
+        let contents = fs::read_to_string(&procfs_cgroup_filepath)
+            .with_context(|| format!("failed to read {:?}", procfs_cgroup_filepath))?;
+
+        // a typical procfs cgroupv2 file will contain only one line
+        // eg: 0::/system.slice/docker-7c7fc86f5f2a609c41c6edd65bd1b64135124a687fa6516f6b177b040d6e3b68.scope
+        // a procfs cgroupv2 file will contains multiple lines
+        // eg:
+        //     2:memory:/daemons
+        //     5:cpuacct,cpu:/daemons
+        // in this case we take the first one arbitrary and log a warning
+        // todo: implement a matrix between metric names and cgroup v1 controllers to be able to select the right cgroup line
+        for line in contents.lines() {
+            let parts: Vec<&str> = line.split(':').collect();
+            if parts.len() >= 3 {
+                if parts[1] != "" {
+                    log::warn!("detected that pid {pid} is cgroup v1 based, selecting arbitrary the cgroup");
```

Comment:
```suggestion
                    log::warn!("detected that process {pid} is managed by cgroup v1, selecting the cgroup arbitrarily");
```
I think you meant "arbitrarily" ^^